app/src/ui/history/compare-branch-list-item.tsx

```diff
@@ -126,12 +127,17 @@ export class CompareBranchListItem extends React.Component<
     return (
       <div className="branches-list-item">
         <Octicon className="icon" symbol={icon} />
-        <div className="name" title={branch.name}>
+        <TooltippedContent
+          className="name"
+          tooltip={branch.name}
+          onlyWhenOverflowed={true}
+          tagName="div"
+        >
           <HighlightText
             text={branch.name}
             highlight={this.props.matches.title}
           />
-        </div>
+        </TooltippedContent>
```

Comment:
Copied logic from `branch-list-item.tsx` for this use case

Before:
![image](https://github.com/desktop/desktop/assets/75402236/9ffb7064-2e90-4c4f-b020-97613f20f42b)

After:
![image](https://github.com/desktop/desktop/assets/75402236/8d43e80c-31e2-4993-9255-59db2db01302)